### PR TITLE
[Twig] Cache template class resolution

### PR DESCRIPTION
**Pull request by smnandre**
Cache template class during rendering.

Non negligeable impact on pages with many components.

Capture d’écran 2024-11-11 à 01 59 30

### CODE REVIEW

File: src/TwigComponent/src/ComponentRenderer.php [hunk 1]
```diff
@@ -24,6 +24,8 @@
  */
 final class ComponentRenderer implements ComponentRendererInterface
 {
+    private array $templateClasses = [];
```
Comment (kbond) [inline src/TwigComponent/src/ComponentRenderer.php:28]: We should reset this state I think.

Comment (smnandre) [reply src/TwigComponent/src/ComponentRenderer.php:28]: Done, and added some (not so) minor optim in Factory again :|

**Comment by smnandre** [comment]
Arf.... again LiveComponent hidden contract .. :( 

Will fix later

**Comment by smnandre** [comment]
Cannot reproduce locally :(

**Comment by smnandre** [comment]
Weird. I think is it related to property accessor but it’s weird this différence between local and CI.

**Comment by Kocal** [comment]
Locally, is your `src/LiveComponent/vendor/symfony/ux-twig-component` points to `src/TwigComponent`?

**Comment by smnandre** [comment]
Ok i confirm... the Live test helper does not simulate real conditions. But i think we can deal with it :) 


This was the change responsible for failures 🤷 


```diff
+       if (!$componentMetadata->isAnonymous()) {
            // set data that wasn't set in mount on the component directly
            foreach ($data as $property => $value) {
                if ($this->propertyAccessor->isWritable($component, $property)) {
                    $this->propertyAccessor->setValue($component, $property, $value);
                    unset($data[$property]);
                }
            }
+        }
```

**Comment by smnandre** [comment]
Fabbot wrong, we can move on with this one :)

**Review by Kocal — APPROVED** [review]

**Pull request merged**